### PR TITLE
Add version checker

### PR DESCRIPTION
**Pull request by axelsheva**

### CODE REVIEW

File: core/repository/socket.ts [hunk 1]
```diff
@@ -73,12 +73,18 @@ export class Socket {
             logger.trace(`[SOCKET][clientConnectToOwnServer] ${ip}`);
 
             const peer = JSON.parse(data);
+            if (peer.version !== config.CORE.VERSION) {
```
Comment (macaqueee) [inline core/repository/socket.ts]: >= MIN_VERSION

File: shared/config/index.ts [hunk 2]
```diff
@@ -149,6 +150,7 @@ class Config {
             IS_HISTORY: process.env.IS_HISTORY === 'TRUE',
             IS_DISABLED_TRANSACTION_CREATION: process.env.IS_DISABLED_TRANSACTION_CREATION === 'TRUE',
             VERSION: '0.0.42',
+            MIN_VERSION: '0.0.42',
```
[the remark this answers is not in the document]
Comment (Verbalist) [reply shared/config/index.ts]: 0.0.1

File: test/core/util/strings.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            },
+            {
+                version: '2.4.42',
+                expected: false,
+            },
+            {
+                version: '0.0.0',
+                expected: false,
+            },
+        ];
```
Comment (Verbalist) [inline test/core/util/strings.test.ts]: add check version 'asd.asd.ads'
and version '123hj'

File: core/util/versionChecker.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export class VersionChecker {
+    private splittedMinVersion: Array<string>;
+
+    constructor(minVersion: string) {
+        this.splittedMinVersion = minVersion.split('.');
+    }
+
+    public isAcceptable(version: string): boolean {
+        const splittedVersion = version.split('.');
```
Comment (Verbalist) [inline core/util/versionChecker.ts]: add validate version, if not valid return false

**Pull request merged**